mlir/lib/Dialect/Rock/IR/RockDialect.cpp

```diff
@@ -697,7 +697,8 @@ static LogicalResult checkGemmSize(Value matrix, Operation *op,
   while (auto transform = raw.getDefiningOp<rock::TransformOp>())
     raw = transform.getInput();
   ShapedType type = raw.getType().cast<ShapedType>();
-  if (!type.hasStaticShape() || type.getSizeInBits() >= fourGbits) {
+  int64_t sizeInBits = type.getNumElements() * type.getElementTypeBitWidth();
```

Comment:
I think `getNumElements()` might crash with non-static shapes (do please double-check) and so we might need a two-level if here.

Reply:
Yep, you are right.